### PR TITLE
[DDCE-1402]:Move user data store from public to protected zone 2

### PR DESCRIPTION
**Pull request by abhijeetardale**
# DDCE-1402

##Bug fix / New feature (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

**Comment by abhijeetardale** [comment]
Linked PR please
https://github.com/hmrc/charities/pull/16
https://github.com/hmrc/app-config-base/pull/4917
https://github.com/hmrc/app-config-staging/pull/7370

File: app/controllers/actions/DataRetrievalAction.scala [hunk 1]
```diff
@@ -19,17 +19,18 @@ package controllers.actions
...
 import play.api.mvc.ActionTransformer
-import repositories.SessionRepository
+import repositories.AbstractRepository
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.HeaderCarrierConverter
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class DataRetrievalActionImpl @Inject()(val sessionRepository: SessionRepository)(
+class DataRetrievalActionImpl @Inject()(val sessionRepository: AbstractRepository)(
```
Comment (anburoach) [inline app/controllers/actions/DataRetrievalAction.scala:28]: this should be sessionRepository: SessionRepository

Comment (abhijeetardale) [reply app/controllers/actions/DataRetrievalAction.scala:28]: this is injected as session using injections
bind(classOf[AbstractRepository]).to(classOf[SessionRepository]).asEagerSingleton()

File: test/controllers/ApplicationBeingProcessedControllerSpec.scala [hunk 2]
```diff
@@ -67,33 +67,32 @@ class ApplicationBeingProcessedControllerSpec extends SpecBase with ImplicitDate
...
 
       val result = controller.onPageLoad()(fakeRequest)
 
       status(result) mustEqual SEE_OTHER
 
       redirectLocation(result).value mustEqual controllers.routes.SessionExpiredController.onPageLoad().url
-      verify(mockUserAnswerRepository, times(1)).get(any())
-      verify(mockUserAnswerRepository, never).delete(any())
+      verify(mockUserAnswerService, times(1)).get(any())(any(), any())
+     // verify(mockUserAnswerService, never).delete(any())
```
Comment (anburoach) [inline test/controllers/ApplicationBeingProcessedControllerSpec.scala:95]: Please remove this as well

Comment (abhijeetardale) [reply test/controllers/ApplicationBeingProcessedControllerSpec.scala]: done and pushed

**Review by anburoach — APPROVED** [review]
Approved

**Pull request merged**